src/common/utils/url-hash-converter/url-hash-converter.ts

```diff
@@ -34,17 +34,19 @@ export const urlHashConverter: UrlHashConverter = {
   essenceFromHash(hash: string, dataCube: DataCube, visualizations: Manifest[]): Essence {
     const hashParts = hash.split('/');
 
-    if (hashParts.length < 3)
-      throw new Error("Wrong url hash structure.");
-
     const isLegacyVersionWithVisualisationPrefix =
```

Comment:
what about the following pseudo code to make intention more explicit:

if hashParts.length < 2 throw error

if hashParts[0] === well known visualisation && hashParts[1] === LEGACY_VERSION
version = ...
encodedModel = ...
visualization = ...
else if hashParts[0] === well known version, e.g 3 or later
version = ...
encodedModel = ...
visualization = undefined
else
throw error

BTW. move conditions into named function